native/src/network.c

```diff
@@ -316,6 +337,11 @@ TCN_IMPLEMENT_CALL(jint, Socket, bind)(TCN_STDARGS, jlong sock,
     TCN_ASSERT(sock != 0);
     TCN_ASSERT(s->sock != NULL);
     rv = (jint)apr_socket_bind(s->sock, a);
+
+    apr_pool_cleanup_register(s->pool, (const void *)s,
```

Comment:
OK, so basically when the Tomcat instance is shut down and all memory pools cleared, the file is removed. Is the socket information retained that long? What will happen when the process crashes, will a restart rebind to the same sock file with `REUSEADDR`?

Reply:
REUSEADDR doesn't appear to work with UDS, alas.

I went digging for best practice, and the thinking seems to be that deleting the file at startup is bad as it masks failures. I've gone with the delete-on-exit route.

The create() function creates the socket pool, the bind() function creates the socket file and registers the socket cleanup, and the destroy() function runs the cleanup. APR itself will run the cleanup on shutdown as long as the java process exits, tomcat itself doesn't need to exit cleanly.

If the java process crashes, the socket file needs to be manually deleted, this appears to be how this is handled elsewhere.

Reply:
OK, can you document the procedure in the case of a crash scenario?

Reply:
It's done.